File: app/src/main/java/com/robivan/simplenote/NoteMapping.java
package com.robivan.simplenote;

import java.util.HashMap;
import java.util.Map;

public class NoteMapping {
    public static class Fields {

        public final static String NAME = "title";
        public final static String DESCRIPTION = "noteText";
        public final static String DATE = "date";
    }

    public static NoteEntity toNoteData(String id, Map<String, Object> doc) {
        NoteEntity answer = new NoteEntity((String) doc.get(Fields.NAME), (String) doc.get(Fields.DESCRIPTION), (long) doc.get(Fields.DATE));
        answer.setId(id);
        return answer;
    }

    public static Map<String, Object> toDocument(NoteEntity noteData) {
        Map<String, Object> answer = new HashMap<>();
        answer.put(Fields.NAME, noteData.getTitle());
        answer.put(Fields.DESCRIPTION, noteData.getNoteText());
        answer.put(Fields.DATE, noteData.getDate());
        return answer;
    }
}

File: app/src/main/java/com/robivan/simplenote/NoteMapping.kt
package com.robivan.simplenote

import java.util.*

object NoteMapping {
    fun toNoteData(id: String?, doc: Map<String?, Any?>): NoteEntity {
        val answer = NoteEntity(
            doc[Fields.TITLE] as String?,
            doc[Fields.NOTE_TEXT] as String?,
            doc[Fields.DATE] as Long
        )
        answer.id = id
        return answer
    }

    fun toDocument(note: NoteEntity): Map<String, Any?> {
        val result: MutableMap<String, Any?> = HashMap()
        result[Fields.TITLE] = note.title
        result[Fields.NOTE_TEXT] = note.noteText
        result[Fields.DATE] = note.date
        return result
    }

    object Fields {
        const val TITLE = "title"
        const val NOTE_TEXT = "noteText"
        const val DATE = "date"
    }
}